orange/src/main/java/com/hartwig/hmftools/orange/algo/linx/LinxBreakendInterpreter.java

```diff
@@ -0,0 +1,140 @@
+
+    public LinxBreakendInterpreter(
+            @NotNull final List<StructuralVariant> structuralVariants,
+            @NotNull final List<LinxSvAnnotation> linxSvAnnotations,
+            @NotNull final EnsemblDataCache ensemblDataCache
+    )
+    {
+        this.structuralVariants = structuralVariants;
+        this.linxSvAnnotations = linxSvAnnotations;
+        this.ensemblDataCache = ensemblDataCache;
+    }
+
+    public LinxBreakend interpret(@NotNull com.hartwig.hmftools.common.linx.LinxBreakend linxBreakend)
+    {
+        ImmutableLinxBreakend.Builder builder = breakendBuilder(linxBreakend);
+        LinxSvAnnotation svAnnotation = linxSvAnnotations.stream()
+                .filter(s -> s.svId() == linxBreakend.svId())
+                .findFirst()
+                .orElse(null);
+
+        if(svAnnotation == null)
+        {
+            LOGGER.warn("No linx sv annotation found for breakend {}, using fallbacks for derived attributes", linxBreakend.id());
+            return builder.build();
+        }
+
+        StructuralVariant sv = structuralVariants.stream()
+                .filter(s -> s.id().equals(svAnnotation.vcfId()))
+                .findFirst()
+                .orElse(null);
+
+        if(sv == null)
+        {
+            LOGGER.warn("No structural variant found for breakend {}, using fallbacks for derived attributes", linxBreakend.id());
+            return builder.build();
+        }
+
+        String chrom = sv.chromosome(linxBreakend.isStart());
+        if(chrom == null)
+        {
+            LOGGER.warn("No chromosome found for breakend {}, using fallbacks for derived attributes", linxBreakend.id());
+            return builder.build();
+        }
+
+        Byte orientation = sv.orientation(linxBreakend.isStart());
+        if(orientation == null)
+        {
+            LOGGER.warn("No orientation found for breakend {}, using fallbacks for derived attributes", linxBreakend.id());
+            return builder.build();
+        }
+
+        double junctionCopyNumber = junctionCopyNumber(linxBreakend, sv, svAnnotation);
+        GeneData geneData = ensemblDataCache.getGeneDataByName(linxBreakend.gene());
+        String chrBand = geneData != null ? geneData.KaryotypeBand : "";
+
+        return builder.chromosome(chrom)
+                .chromosomeBand(chrBand)
+                .type(LinxBreakendType.valueOf(sv.type().name()))
+                .orientation(orientation)
+                .junctionCopyNumber(junctionCopyNumber)
+                .build();
+    }
+
+    @VisibleForTesting
+    static double junctionCopyNumber(
+            @NotNull com.hartwig.hmftools.common.linx.LinxBreakend linxBreakend,
+            @NotNull StructuralVariant sv,
+            @NotNull LinxSvAnnotation svAnnotation
+    )
+    {
+        if(sv.type() == StructuralVariantType.SGL && linxBreakend.regionType() == com.hartwig.hmftools.common.gene.TranscriptRegionType.IG)
+        {
+            return 0D;
+        }
```

Comment:
@DFKoetsier this is what I meant by SGL & IG. Was just a quick observation from running on some tests that cases with 0 had these attributes. Haven't worked out why or if comprehensive.

Reply:
You suggested elsewhere that if the simple average looks reasonable perhaps we could stick with it. I'm inclined now to simplify the code and just use the average. I've found another example of the min=0 special case where it behaves differently, and I don't want to further complicate.